pkg/operator/ceph/cluster/mon/service.go

```diff
@@ -52,6 +52,18 @@ func (c *Cluster) createService(mon *monConfig) (string, error) {
 		svcDef.Spec.ClusterIP = v1.ClusterIPNone
 	}
 
+	// allows operators the ability to expose their rook-ceph mons externally
+	// by supplying their own `loadBalancerIP`. Since this field can only be used
+	// with `type` LoadBalancer, we automatically convert the service definition to use
+	// `LoadBalancer` instead of `ClusterIP` (the default)
+	if c.spec.Network.LoadBalancerService {
```

Comment:
Instead of modifying the service type here (and getting one per mon), what if you created a single load balancer service? The service could use labels such that it would apply to all the mon pods. This would give you the behavior of a single endpoint that lets you round robin between all the mons. You would also not be affected by mons internally being failed over by rook. I believe this single endpoint approach will work for the ceph clients.